### PR TITLE
[Fix] Fix the support for ROCm

### PR DESCRIPTION
**Pull request by xiabo123**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

This PR enables HIP as a backened for pytorch only.

## Modification

1. conv2d_gradfix.py :Torch version comparison, if the version number has rc or a, there is an error.Conv calls the conversion from cudnn to miopen.
2. corner_pool.py：Torch version comparison, if the version number has rc or a, there is an error.
3. bias_act_cuda.cu/upfirdn2d_kernel.cu : Torch hipify transcoding rule is missing.
4. filtered_lrelu.cu ： cuda : __shfl_xor_sync -> rocm : __shfl_xor
5. test_filtered_lrelu.py ： Test case, add rocm judgment.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/ops/conv2d_gradfix.py [hunk 1]
```diff
@@ -283,15 +283,36 @@ def forward(ctx, grad_output, input):
...
+                    from torch.utils.cpp_extension import ROCM_HOME
+                    if ((torch.version.hip is not None)
+                            and (ROCM_HOME is not None)):
+                        is_rocm_pytorch = True
+                except ImportError:
+                    pass
+
+                name = ''
+                flags = []
+
```
Comment (zhouzaida) [inline mmcv/ops/conv2d_gradfix.py:297]:
```suggestion
```

File: mmcv/ops/conv2d_gradfix.py [hunk 2]
```diff
@@ -283,15 +283,36 @@ def forward(ctx, grad_output, input):
...
+                if is_rocm_pytorch:
```
Comment (zhouzaida) [inline mmcv/ops/conv2d_gradfix.py:298]:
```suggestion
                if is_rocm_pytorch():
```

Comment (zhouzaida) [reply mmcv/ops/conv2d_gradfix.py:298]: We can reuse this function from mmengine.

```python
from mmengine.utils.dl_utils import is_rocm_pytorch
```

File: tests/test_ops/test_filtered_lrelu.py [hunk 3]
```diff
@@ -114,10 +114,22 @@ def test_filtered_lrelu_cpu(self):
...
+        pass
+
+    is_skipif = True
+    if is_rocm_pytorch:
+        is_skipif = False
+    else:
+        is_skipif = (not torch.cuda.is_available() or
+                     digit_version(torch.version.cuda) < digit_version('10.2'))
+
+    @pytest.mark.skipif(is_skipif, reason='requires cuda>=10.2')
```
Comment (zhouzaida) [inline tests/test_ops/test_filtered_lrelu.py:132]:
```suggestion
    @pytest.mark.skipif(
        not torch.cuda.is_available() or is_rocm_pytorch() or
        or digit_version(torch.version.cuda) < digit_version('10.2'),
        reason='requires cuda>=10.2')
```

File: mmcv/ops/conv2d_gradfix.py (hunk 2 above)
Comment (xiabo123) [reply mmcv/ops/conv2d_gradfix.py]: There is no is_rocm_pytorch in python3.7

File: tests/test_ops/test_filtered_lrelu.py (hunk 3 above)
Comment (xiabo123) [reply tests/test_ops/test_filtered_lrelu.py]: Thank you for your testimony. However, on the rocm end, the value of "torch. version. cuda" is none, digit_ Version may be incorrect.

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**